docs/develop/smart-contracts/outposts/osmosis.md

```diff
@@ -0,0 +1,50 @@
+---
+sidebar_position: 1
+---
+
+# Osmosis Outpost
+
+[The Osmosis Outpost](https://app.evmos.org/dapps/defi/osmosis) is an Outpost that leverages
+the [IBC Transfer Extension](https://docs.evmos.org/develop/smart-contracts/evm-extensions/ibc-transfer)
+to interact with the Osmosis [IBC-hooks middleware](https://github.com/osmosis-labs/osmosis/tree/main/x/ibc-hooks) 
+and the Osmosis SwapRouter and Cross Chain Swaps contract.
+It provides an easy-to-use interface for executing token swaps between Osmosis and Evmos 
+and routing them back to the Evmos chain. 
+
+## Key Components
+- **`memo` field**: An extra JSON string attached to IBC Transfer packets. This additional information helps
+  the IBC Hooks to identify the user and the swap being performed.
+
+
+- **Osmosis IBC Hooks Middleware**: Middleware on Osmosis that interprets the memo field for specific actions.
+
+
+- **Osmosis SwapRouter**: A contract that manages routes to pools on Osmosis, defining how tokens are swapped.
+
+
+- **Osmosis Cross-Chain Swap V1 (XCS)**: Handles processing of the memo field and executes token swaps or 
+routes tokens to their destination chains. Evmos has its own XCS contract deployed on the Osmosis chain. You can find it
+[here](https://celatone.osmosis.zone/osmosis-1/contracts/osmo18rj46qcpr57m3qncrj9cuzm0gn3km08w5jxxlnw002c9y7xex5xsu74ytz)
+
+## How it works
+
+Users can interact with the Osmosis Outpost directly using the Osmosis instant dApp
+from the [Evmos dApp store](https://app.evmos.org/dapps/defi/osmosis). The dApp will automatically build
+the correct `memo` field for the user and send the IBC transfer to the Osmosis chain. The possible actions are:
+
+1. **Swap Osmosis for Evmos** - Send your Osmosis tokens to the Osmosis Outpost and receive Evmos tokens in return.
+
+2. **Swap Evmos for Osmosis** - Send your Evmos tokens to the Osmosis Outpost and receive Osmosis tokens in return.
```

Comment:
We only have registered token pairs for singly wrapped Coins. The Coins on Osmosis will be doubly wrapped as they had to first come from the origin chain to Osmosis. V2 contract would have solved this as it will use PFM to route it back to the origin chain and then send it to Evmos but they are not confident in v2 at the moment and don't really use it.